schematic/store/synapse.py

```diff
@@ -100,23 +101,46 @@ def __init__(
 
         self._query_fileview()
 
+    def _purge_synapse_cache(self):
+        '''
+        Purge synapse cache if it exceeds 19.5 GB
+        '''
+        # try clearing the cache
+        # scan a directory and check size of files
+        root_dir = self.syn.cache.cache_root_dir
+        cache = self.syn.cache
+        if os.path.exists(root_dir):
+            nbytes = get_dir_size(root_dir)
+            # if 19.5 GB has already been taken, purge cache before an hour
+            if nbytes >= 20937965568:
+                an_hour_earlier = datetime.strftime(datetime.utcnow()- timedelta(hours = 1), '%s')
```

Comment:
I recommended an hour, but is this reasonable?  How fast did you encounter the storage space overfill again?

Reply:
Note: I have already updated it to 15 min and also clear cache when it reaches 7GB